bot/cogs/filtering.py

```diff
@@ -237,8 +250,9 @@ def mod_log(self) -> ModLog:
         Only matches words with boundaries before and after the expression.
         """
         for regex_pattern in WORD_WATCHLIST_PATTERNS:
-            if regex_pattern.search(text):
-                return True
+            match = regex_pattern.search(text)
+            if match:
+                return match  # match objects always have a boolean value of True
```

Comment:
How about `return match[0]` here? Would have to explicitly check  `triggered is not False` though (perhaps better to return `None` instead of `False` in that case).

Reply:
I return the match because I still need to access the the match start and end positions to get the `Location:` information, here:
`surroundings = m.string[max(m.start() - 10, 0): m.end() + 10]`